pkg/cache/internal/informers.go

```diff
@@ -92,6 +107,25 @@ type tracker struct {
 	Metadata     map[schema.GroupVersionKind]*Cache
 }
 
+// GetOptions provides configuration to customize the behavior when
+// getting an informer.
+type GetOptions struct {
+	// Async instructs Get() to return the informer immediately,
+	// without waiting for its cache to sync.
+	Async bool
```

Comment:
```suggestion
	WaitForInformerSynced bool
```
Its not very clear to me what `Async` means, maybe its just me.

Reply:
This is moved to https://github.com/kubernetes-sigs/controller-runtime/pull/2371

I'll add the suggestion in there.

Reply:
Wait... Async was a separate suggestion. Leaving `DoNotBlockUntilSynced` for now, we can debate naming on the new PR.

One thing to note is that `false` should mean "block until ready" in order to maintain backwards compatible behavior when no args are applied.

Reply:
Closing this, since non-blocking was moved to #2371